app/lib/efile/ny/it201.rb

```diff
@@ -3,63 +3,68 @@ module Ny
     class It201
       attr_reader :lines
 
-      def initialize(year:, filing_status:, claimed_as_dependent:, dependent_count:, input_lines:, it227:)
+      def initialize(year:, filing_status:, claimed_as_dependent:, dependent_count:, input_lines:, it213:, it214:, it215:, it227:, value_access_tracker:)
```

Comment:
I made a few changes here:

- Rename `@collaborators` to `@inputs` (I think inputs is a more straightforward word in this context)
- Change ValueAccessTracker so it's created earlier in the timeline, rather than created in this initialize (this is a 'worse is better' approach to initialization - make your callers do more work and then they are confident in the implementation)
- Unify all TaxFormLine classes into one, since their behavior changes are limited now to factory methods

Reply:
I think I agree with everything but creating `ValueAccessTracker` outside -- it is **extremely** an implementation detail of the tax calculator / tax form line classes, and the less things that need to be aware of it the better

I understand the light ickyness in assigning a value_access_tracker to already-instantiated lines, but it seems like a good tradeoff to keep references to it isolated

Reply:
Interesting, yeah. I'm OK with restoring the `value_access_tracker=` stuff.

It's true that value access tracking is an implementation detail. I wonder if there's yet another option waiting for us which accomplishes even more of our shared goals.

Reply:
Restored